internal/cachecontroller/cache_controller.go

```diff
@@ -87,15 +90,27 @@ func (c *InMemoryCacheController) DetermineInvalidation(
 		return entry.LastModified
 	}
 
-	lastModified, err, _ := c.sf.Do(storeID, func() (interface{}, error) {
-		return c.findChangesAndInvalidate(ctx, storeID)
-	})
-
-	if err != nil {
-		return time.Time{}
+	c.currentInvalidationMutext.Lock()
+	_, present := c.currentInvalidationMap[storeID]
+	if !present {
+		c.currentInvalidationMap[storeID] = struct{}{}
 	}
+	c.currentInvalidationMutext.Unlock()
+	if !present {
+		span.SetAttributes(attribute.Bool("checkInvalidation", true))
+		// if the cache cannot be found, we want to invalidate entries in the background
+		// so that it does not block the answer path.
+		go func() {
+			_ = c.findChangesAndInvalidate(ctx, storeID)
```

Comment:
Should this be the same context? Its a background thing, and if the Check resolves first, the context will be cancelled, and we actually want to continue processing.

Reply:
updated to use context without timeout.